ReactiveUI/RxApp.cs

```diff
@@ -69,7 +69,15 @@ static RxApp()
                 LogHost.Default.Warn("ReactiveUI acts differently under a test runner, see the docs\n");
                 LogHost.Default.Warn("for more info about what to expect");
 
-                _MainThreadScheduler = CurrentThreadScheduler.Instance;
+                // This sets _UnitTestMainThreadScheduler but only on the current thread. It doesn't set 
+                // _MainThreadScheduler if it's already been set. And the registration code generally sets this to a
+                // WaitForDispatcherScheduler _before_ we get to this line.
+                MainThreadScheduler = CurrentThreadScheduler.Instance;
+                // This is why we _also_ have to set this field so that the default scheduler while running in unit
+                // tests is the immediate scheduler. This is particularly important on xunit 2 where every test 
+                // might be run in its own thread. We found that in those cases, _UnitTestMainThreadScheduler was 
+                // null and we would get a bunch of tests using the WaitForDispatcherScheduler for major loss.
+                _MainThreadScheduler = MainThreadScheduler;
```

Comment:
I believe that would cause a problem when running in unit tests. If a unit test invokes an async operation, `_UnitTestMainThreadScheduler` is null on that other thread. This is because `_UnitTestMainThreadScheduler` is declared to be `ThreadStatic`. It's "static", but only on the current thread.

So if you don't set `_MainThreadScheduler` when it's null, then any async operation will have both `_UnitTestMainThreadScheduler` null and `_MainThreadScheduler` null.

Unless I'm missing something about your proposal.